bundles/org.openhab.binding.haywardomnilogic/src/main/java/org/openhab/binding/haywardomnilogic/internal/discovery/HaywardDiscoveryService.java

```diff
@@ -150,6 +150,15 @@ public synchronized void mspConfigDiscovery(String xmlResponse) {
         final List<String> colorLogicProperty1 = bridgehandler.evaluateXPath("//Backyard//ColorLogic-Light/Type/text()",
                 xmlResponse);
 
+        final List<String> colorLogicProperty2 = bridgehandler
+                .evaluateXPath("//Backyard//ColorLogic-Light/V2-Active/text()", xmlResponse);
+
+        for (int i = 0; i < colorLogicProperty2.size(); i++) {
+            if (colorLogicProperty1.get(i).equals("COLOR_LOGIC_UCL") && colorLogicProperty2.get(i).equals("yes")) {
```

Comment:
Is it obvious that `colorLogicProperty1 ` and `colorLogicProperty2` have the same number of entries ?

Reply:
colorLogicProperty2.size will always be <= to the colorLogicProperty1.size

In other words, colorLogicProperty2 only exists if it is the newer V2 light.

Reply:
Ok